app/common/runtimeTypes/ipc/address.ts

```diff
@@ -9,26 +9,41 @@ export const GenerateAddressParams = t.intersection([
   }),
   t.partial({
     label: t.string,
-    amount: t.number,
+    requestedAmount: t.number,
     expirationDate: t.number
   })
 ], "GenerateAddressParams")
 export type GenerateAddressParams = t.TypeOf<typeof GenerateAddressParams>
 
 export const GenerateAddressSuccess = t.type({
   kind: t.literal("SUCCESS"),
-  keyPath: KeyPath,
-  address: t.string,
-  creationDate: t.number
+  finalKey: ExternalFinalKey
 }, "GenerateAddressSuccess")
-
 export type GenerateAddressSuccess = t.TypeOf<typeof GenerateAddressSuccess>
 
 export const generateAddressErrors = {
+  WRONG_PARAMS: t.literal("WRONG_PARAMS"),
+  NO_UNLOCKED_WALLET: t.literal("NO_UNLOCKED_WALLET"),
+  WRONG_ACCOUNT: t.literal("WRONG_ACCOUNT"),
+  NEGATIVE_AMOUNT: t.literal("NEGATIVE_AMOUNT"),
+  PAST_EXPIRATION_DATE: t.literal("PAST_EXPIRATION_DATE"),
+  TOO_FAR_EXPIRATION_DATE: t.literal("TOO_FAR_EXPIRATION_DATE"),
+  WALLET_UPDATE_FAILURE: t.literal("WALLET_UPDATE_FAILURE"),
+  WALLET_STORE_FAILURE: t.literal("WALLET_STORE_FAILURE"),
+  ADDRESS_ENCODING_FAILURE: t.literal("ADDRESS_ENCODING_FAILURE"),
```

Comment:
`WALLET_UPDATE_FAILURE`, `WALLET_STORE_FAILURE` and `ADDRESS_ENCODING_FAILURE` errors are not being used in the handler

Reply:
I removed them. If we need to consider any of these errors in the future, we could included them later.